common/v2/features/Dashboard/components/WalletBreakdown/WalletBreakdownView.tsx

```diff
@@ -164,15 +164,61 @@ export default function WalletBreakdownView({
   const [selectedAssetIndex, setSelectedAssetIndex] = useState(0);
   const [previousBalances, setPreviousBalances] = useState<Balance[]>([]);
 
-  if (balances.length !== previousBalances.length) {
-    setSelectedAssetIndex(0);
-    setPreviousBalances(balances);
+  const SMALLEST_CHART_SHARE_SUPPORTED = 0.03; // 3%
+
+  const NUMBER_OF_ASSETS_DISPLAYED = 4;
+
+  /* Construct a finalBalances array which consists of top X assets and a otherTokensAsset
```

Comment:
Please refactor L171 to 210 to use pure functions instead of conditionals and re-assignments